plugins/woocommerce/client/blocks/assets/js/blocks/product-details/edit.tsx

```diff
@@ -109,24 +45,115 @@ const useIsInvalidQueryLoopContext = ( clientId: string, postType: string ) => {
 const Edit = ( { clientId, context }: ProductDetailsEditProps ) => {
 	const blockProps = useBlockProps();
 
+	const product = useSelect(
+		( select ) => {
+			if ( ! context.postId ) {
+				return null;
+			}
+			const { getProduct } = select( productsStore );
+			return getProduct( Number( context.postId ) );
+		},
+		[ context.postId ]
+	);
+
+	const isInnerBlockOfSingleProductBlock = useSelect(
+		( select ) => {
+			const blocks = select(
+				blockEditorStore
+				// @ts-expect-error - getBlockParentsByBlockName is not typed
+			).getBlockParentsByBlockName(
+				clientId,
+				'woocommerce/single-product'
+			);
+			return blocks.length > 0;
+		},
+		[ clientId ]
+	);
+
+	const template = useMemo( () => {
+		return getTemplate( product, {
+			isInnerBlockOfSingleProductBlock,
+		} );
+	}, [ product, isInnerBlockOfSingleProductBlock ] );
+
 	const { hasInnerBlocks, wasBlockJustInserted } = useSelect(
 		( select ) => {
+			// @ts-expect-error - getBlocks is not typed
 			const blocks = select( blockEditorStore ).getBlocks( clientId );
+			const innerBlocks = blocks.length > 0;
+			const blockJustInserted =
+				// @ts-expect-error - wasBlockJustInserted is not typed
+				select( blockEditorStore ).wasBlockJustInserted( clientId );
+
 			return {
-				hasInnerBlocks: blocks.length > 0,
-				wasBlockJustInserted:
-					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
-					// @ts-ignore method exists but not typed
-					select( blockEditorStore ).wasBlockJustInserted( clientId ),
+				hasInnerBlocks: innerBlocks,
+				wasBlockJustInserted: blockJustInserted,
 			};
 		},
 		[ clientId ]
 	);
 
+	const { accordionItemClientId } = useSelect(
+		( select ) => {
+			if ( ! wasBlockJustInserted ) {
+				return {
+					accordionItemClientId: null,
+				};
+			}
+
+			const productSpecificationClientIds = select(
+				blockEditorStore
+				// @ts-expect-error - getBlocksByName is not typed
+			).getBlocksByName( 'woocommerce/product-specifications' );
+
+			const productDetailsDescendants =
+				// @ts-expect-error - getClientIdsOfDescendants is not typed
+				select( blockEditorStore ).getClientIdsOfDescendants(
+					clientId
+				);
+
+			const productSpecificationClientId =
+				productSpecificationClientIds.find( ( id: string ) =>
+					productDetailsDescendants.includes( id )
+				);
```

Comment:
Consider using `getInnerBlockByName` from `@woocommerce/utils`, I think it would simplify it a bit:

```suggestion
			const productDetailsBlock = select(
				blockEditorStore
				// @ts-expect-error - getBlocksByName is not typed
			).getBlock( clientId );
			
			const productSpecificationClientId = getInnerBlockByName(
			    productDetailsBlock,
			    'woocommerce/product-specifications'
			)?.clientId;
```

or even simplify further when looking at further code:

```suggestion
			const productDetailsBlock = select(
				blockEditorStore
				// @ts-expect-error - getBlocksByName is not typed
			).getBlock( clientId );
			
			const productSpecificationBlock = getInnerBlockByName(
			    productDetailsBlock,
			    'woocommerce/product-specifications'
			);
			
			const accordionClientId = getInnerBlockByName(
			    productSpecificationBlock,
			    'woocommerce/accordion-item'
			)?.clientId;
```

Reply:
I addressed all your feedback regarding `useSelect` with c28b3bb56a428124a88178dd768ec689c5cad5da. Thanks! 🙇